node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -43,6 +44,10 @@ pub struct OrderingProvider {
 	///
 	/// We need this to clean up `included_candidates` on `ActiveLeavesUpdate`.
 	candidates_by_block_number: BTreeMap<BlockNumber, HashSet<CandidateHash>>,
+	/// Hash of the latest relay block observed by the provider. It's assumed that
+	/// all ancestors of this block are already processed, i.e. we have saved corresponding
+	/// included candidates.
+	last_observed_block: Hash,
```

Comment:
I would say an LRU of 20 heads or something, would provide decent support for forks .... but this is only an optimization, as mentioned below we should make sure to not go back to genesis, just because some head was missing in the LRU.